### PR TITLE
[JENKINS-58038] Use Annotation-Indexer instead of Reflections

### PR DESCRIPTION
**Pull request by AbhyudayaSharma**
Reflections was causing a dependency conflict with the Guava from
Jenkins Core (see: https://travis-ci.org/jenkinsci/configuration-as-code-plugin/jobs/546110907#L878)

@jenkinsci/gsoc2019-role-strategy

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
Thanks for the investigation @AbhyudayaSharma! Maybe it would be even better to use [sezpoz](https://github.com/jglick/sezpoz) since it is already included into JTH. 

A Jenkins JIRA ticket would be also much appreciated since the previous JTH version has been already released

**Comment by casz** [comment]
Green built in downstream PR: https://github.com/jenkinsci/configuration-as-code-plugin/pull/921

**Comment by oleg-nenashev** [comment]
My plan is to merge and release it on the evening if there is no negative feedback

File: src/main/java/jenkins/benchmark/jmh/BenchmarkFinder.java [hunk 1]
```diff
@@ -1,40 +1,26 @@
...
-        Set<Class<?>> benchmarkClasses = reflections.getTypesAnnotatedWith(JmhBenchmark.class);
-        benchmarkClasses.forEach(clazz -> {
-            JmhBenchmark annotation = clazz.getAnnotation(JmhBenchmark.class);
-            if (Objects.nonNull(annotation)) {
-                optionsBuilder.include(clazz.getName() + annotation.value());
-            }
-        });
+    public static void findBenchmarks(ChainedOptionsBuilder optionsBuilder) {
+        for (IndexItem<JmhBenchmark, Object> item : Index.load(JmhBenchmark.class, Object.class)) {
+            optionsBuilder.include(item.className() + item.annotation().value());
```
Comment (jglick) [inline src/main/java/jenkins/benchmark/jmh/BenchmarkFinder.java]: If you have to call `IndexItem.className` you are not using SezPoz correctly. If you do not care about lazy loading and just want to enumerate classes marked with a certain annotation, then use `org.jenkins-ci:annotation-indexer` (also already in the classpath).

**Pull request merged**